### PR TITLE
[6.3] FIX test_permissions

### PR DESCRIPTION
**Pull request by ldjebran**
```console
(sat-6.3) dl@DL:~/projects/redhat/robottelo$ py.test -v tests/foreman/api/test_permission.py::PermissionTestCase
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dl/.pyenv/versions/2.7.13/envs/sat-6.3/bin/python2.7
cachedir: .cache
rootdir: /home/dl/projects/redhat/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, cov-2.3.1
collected 3 items 
2017-06-29 20:30:11 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search_by_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search_by_resource_type <- robottelo/decorators/__init__.py PASSED

============================================== 3 passed in 321.64 seconds ==============================================
```
also fixed one affected test test_positive_subscribe_host_with_restricted_user_permissions, but fail because of bug  https://bugzilla.redhat.com/show_bug.cgi?id=1465468

### CODE REVIEW

**Comment by ldjebran** [comment]
This PR also fix UI adgroup tests
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_adusergroup.py -v -k "test_positive_add_katello_role or test_positive_update_external_user_roles"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 11 items 
2017-06-30 18:01:49 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-06-30 18:01:49 - conftest - DEBUG - Collected 11 test cases

2017-06-30 18:01:49 - conftest - DEBUG - Deselected test tests.foreman.ui.test_adusergroup.test_positive_update_external_roles due to WONTFIX


tests/foreman/ui/test_adusergroup.py::ActiveDirectoryUserGroupTestCase::test_positive_add_katello_role PASSED
tests/foreman/ui/test_adusergroup.py::ActiveDirectoryUserGroupTestCase::test_positive_update_external_user_roles PASSED

================================================== 9 tests deselected ==================================================
======================================= 2 passed, 9 deselected in 376.56 seconds =======================================
```

**Comment by sghai** [comment]
ACK

File: robottelo/constants.py [hunk 1]
```diff
@@ -836,17 +846,20 @@
         'sync_products',
         'export_products',
     ],
+    'Katello::Subscription': [
+        'view_subscriptions',
+        'attach_subscriptions',
+        'unattach_subscriptions',
+        'import_manifest',
+        'delete_manifest',
+
```
Comment (oshtaier) [inline robottelo/constants.py]: redundant line

**Comment by oshtaier** [comment]
Is it intended and confirmed changes to permissions? Did they completely remove things related to Ansible?

**Comment by ldjebran** [comment]
put do not merge for investigations

**Comment by sghai** [comment]
@oshtaier : ansible roles related stuff is completely out of 6.3. So that change is expected. A email thread was already there in internal list referencing same. 

Ssh-key, oscap-tailoring files and virt-who related permissions are expected too due to new features.

Thank you for confirming all these changes.

**Comment by ldjebran** [comment]
```console
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 3 items 
2017-07-06 16:14:14 - conftest - DEBUG - Found WONTFIX in decorated tests ['1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1199150', '1204686', '1267224', '1221971', '1103157', '1230902', '1214312', '1079482']

2017-07-06 16:14:14 - conftest - DEBUG - Collected 3 test cases


tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search_by_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/api/test_permission.py::PermissionTestCase::test_positive_search_by_resource_type <- robottelo/decorators/__init__.py PASSED

================================================== 0 tests deselected ==================================================
============================================== 3 passed in 313.21 seconds ==============================================
```

**Pull request merged**